lib/events/dynamoevents/dynamoevents_test.go

```diff
@@ -74,71 +69,82 @@ func (s *suiteBase) SetUpSuite(c *check.C) {
 		Clock:        fakeClock,
 		UIDGenerator: utils.NewFakeUID(),
 	}, backend)
-	c.Assert(err, check.IsNil)
-	s.log = log
-	s.EventsSuite.Log = log
-	s.EventsSuite.Clock = fakeClock
-	s.EventsSuite.QueryDelay = time.Second * 5
-}
+	require.NoError(t, err)
 
-func (s *suiteBase) SetUpTest(c *check.C) {
-	err := s.log.deleteAllItems(context.Background())
-	c.Assert(err, check.IsNil)
+	return &dynamoContext{
+		log: log,
+		suite: test.EventsSuite{
+			Log:        log,
+			Clock:      fakeClock,
+			QueryDelay: time.Second * 5,
+		},
+	}
 }
 
-func (s *suiteBase) TearDownSuite(c *check.C) {
-	if s.log != nil {
-		if err := s.log.deleteTable(context.Background(), s.log.Tablename, true); err != nil {
-			c.Fatalf("Failed to delete table: %#v", trace.DebugReport(err))
-		}
+func (tt *dynamoContext) Close() error {
+	if tt.log != nil {
+		return tt.log.deleteTable(context.Background(), tt.log.Tablename, true)
 	}
+
+	return nil
 }
 
-type DynamoeventsSuite struct {
-	suiteBase
+func TestPagination(t *testing.T) {
+	tt := setupDynamoContext(t)
```

Comment:
The guideline I've been giving is that constructors and other test helpers that take a `*testing.T` should register any necessary cleanups.

So in this case, move the `t.Cleanup()` call into `setupDynamoContext()`, then the callers don't need to worry about it.